util/run-git-clang-format.py

```diff
@@ -0,0 +1,367 @@
+    IGNORE_ENV_VAR_NAME: str = "GEM5_GIT_CLANG_FORMAT_IGNORE"
+
+    PROGRAM_DESCRIPTION_LINES: list[str] = [
+        "Run `git-clang-format` on the current staged changes.",
+        "",
+        "Changes in individual files can be ignored using the `--ignore` "
+        "option or by setting the environment variable "
+        f"`{IGNORE_ENV_VAR_NAME}` to a colon-separated list of "
+        f"files. Set environment variable `{DISABLE_ENV_VAR_NAME}` "
+        "to disable `git-clang-format` globally."
+        "",
+        "When running from `pre-commit`, specify the `--pre-commit` ",
+        "option to suppress output and return appropriate exit codes.",
+    ]
+    PROGRAM_DESCRIPTION: str = "\n".join(PROGRAM_DESCRIPTION_LINES)
+
+    parser = argparse.ArgumentParser(
+        description=PROGRAM_DESCRIPTION,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--pre-commit",
+        action="store_true",
+        help="Indicates the script is being run from pre-commit.",
+    )
+    parser.add_argument(
+        "--ignore",
+        "-i",
+        action="append",
+        default=list(),
+        help="Ignore this file.",
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Print the diff to stdout but do not modify any files.",
+    )
+    parser.add_argument(
+        "--verbose",
+        "-v",
+        action="store_true",
+        help="Output verbose debugging messages to stderr.",
+    )
+    args = parser.parse_args()
+
+    # Set up logging
+    #
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(levelname)7s: %(message)s",
+        stream=sys.stderr,
+    )
+    logger = logging.getLogger(__name__)
+    if args.pre_commit:
+        logger.setLevel(logging.CRITICAL + 1)
+    elif args.verbose:
+        logger.setLevel(logging.DEBUG)
+
+    # Get the repository root directory
+    #
+    res = run_silent_exit(
+        "git", "rev-parse", "--show-toplevel", silent_exit=args.pre_commit
+    )
+    if res is None:
+        logger.error("Not a git repository, or git tools not available.")
+        sys.exit(2)
+    repo_root = Path(res.stdout.strip()).resolve()
+    logger.debug("Repository root: '%s'", repo_root)
+    make_ignore_path = functools.partial(
+        make_ignore_path_rel, repo_root=repo_root
+    )
+
+    # Generate a list of ignored files, and convert to git pathspecs.
+    #
+    # The list combines ignored files specified on the command line
+    # and via the environment variable. Missing files are removed.
+    # Finally, `git` is queried to get a list of staged files, and
+    # ignored files are removed to generate a git pathspec for the
+    # operation. This is necessary because `git-clang-format` does not
+    # support an ignored file list directly.
+    #
+    # Also check the environment variable for a global disable.
+    #
+    if DISABLE_ENV_VAR_NAME in os.environ:
```

Comment:
What happens if I set

`export GEM5_NO_GIT_CLANG_FORMAT=0`

An user might intuitively believe git-clang-format would be enabled. Shall we check if it is a non zero value or would that be a bit too tricky?

Reply:
Makes sense! I believe updating the argparse description would be enough. Afterall, if the script does not get disabled and a style inconsistency is found, it will be promptly signalled to the committer. It shouldn't take long for a user to debug the issue